packages/flutter/lib/src/material/segmented_button.dart

```diff
@@ -0,0 +1,608 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:math' as math;
+
+import 'package:flutter/rendering.dart';
+import 'package:flutter/widgets.dart';
+import 'button_style.dart';
+import 'color_scheme.dart';
+import 'icons.dart';
+import 'material_state.dart';
+import 'segmented_button_theme.dart';
+import 'text_button.dart';
+import 'theme.dart';
+
+// TODO(darrenaustin): do we really need this?
+const double _kMinSegmentedButtonHeight = 28.0;
+
+/// Data describing a segment of a [SegmentedButton].
+class SegmentData<T> {
```

Comment:
This seems like a somewhat generic name. SegmentedButtonSegment would be horrible, albeit more specific. Maybe just `ButtonSegment`?